src/Microsoft.Identity.Client/ApiEvent.cs

```diff
@@ -83,5 +109,20 @@ public bool WasSuccessful
             set => this[ConstWasSuccessful] = value.ToString().ToLower();
             get => this[ConstWasSuccessful] == true.ToString().ToLower();
         }
+
+        public string CorrelationId
+        {
+            set => this[ConstCorrelationId] = value;
```

Comment:
Hmm, normally whenever the thing is an ID, we shouldn't make any transformation to it. But lucky that in this case, it is a Guid [we generate](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/dev/src/Microsoft.Identity.Client/ClientApplicationBase.cs#L228), and [in C#, a Guid is in lowercase by default](https://msdn.microsoft.com/en-us/library/560tzess(v=vs.110).aspx). So we don't need to do anything, it just works.

PS: In future our API may allow caller to provide a CorrelationID ( @kpanwar is that ever possible?) . Then we will need some extra work to ensure they won't provide uppercase Guid. But that is a different story.